pkgs/by-name/ch/checkmk-agent/package.nix

```
@@ -0,0 +1,233 @@
+  nativeBuildInputs = [
+    rpmextract
+    gzip
+    makeWrapper
+  ];
+
+  dontBuild = true;
+
+  # unpack rpm into distinct source root (`src`)
+  unpackCmd = ''
+    mkdir tmp
+    cd tmp
+    rpmextract $curSrc
+    mv opt/omd/versions/${major}.cre/share/check_mk/agents/check-mk-agent-${version}.noarch.rpm ..
+    cd ..
+    rm -rf tmp
+
+    mkdir src
+    cd src
+    rpmextract ../check-mk-agent-${version}.noarch.rpm
+    gunzip var/lib/cmk-agent/cmk-agent-ctl.gz
+    cd ..
+  '';
+  sourceRoot = "src";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/lib/systemd/system
+    cp usr/bin/* var/lib/cmk-agent/cmk-agent-ctl $out/bin
+    chmod +x $out/bin/cmk-agent-ctl
+
+    ${lib.strings.optionalString postfixSupport ''
+      substitute "$out/bin/check_mk_agent" "$out/bin/check_mk_agent" \
+        --replace /usr/sbin/ssmtp ${postfix}
+    ''}
+
+    for cfg in var/lib/cmk-agent/scripts/super-server/0_systemd/*.{service,socket,fallback}; do
+      substitute "$cfg" "$out/lib/systemd/system/$(basename $cfg)" \
+        --replace /usr/bin/waitmax $out/bin/waitmax \
+        --replace /usr/bin/check_mk_agent $out/bin/check_mk_agent \
+        --replace /usr/bin/check_mk_caching_agent $out/bin/check_mk_caching_agent \
+        --replace /usr/bin/cmk-agent-ctl $out/bin/cmk-agent-ctl \
+        --replace /usr/bin/mk-job $out/bin/mk-job
+    done
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    wrapProgram $out/bin/cmk-agent-ctl \
+      --set PATH $out/bin \
+      --set MK_TMPDIR /tmp \
+      --set MK_VARDIR /var/lib/check_mk_agent \
+      --set MK_CONFDIR /etc/check_mk
+    wrapProgram $out/bin/mk-job \
+    --set PATH ${
+      lib.makeBinPath [
+        coreutils
+        gnugrep
+        procps
+      ]
+    }:$out/bin
+
+    wrapProgram $out/bin/check_mk_caching_agent \
+      --set PATH ${
+        lib.makeBinPath [
+          coreutils
+          util-linux
+        ]
+      }:$out/bin
+    wrapProgram $out/bin/check_mk_agent \
+      --set MK_LIBDIR ${toString lib_dir} \
+      --set MK_BIN $out \
+      --set MK_CONFDIR /etc/check_mk \
+      --set MK_VARDIR /var/lib/check_mk_agent \
+      --set MK_TMPDIR /tmp \
+      --set PATH ${
+        lib.makeBinPath (
+          [
+            procps
+            coreutils
+            findutils
+            perl
+            systemd
+            python3
+            gnugrep
+            gawk
+            gnused
+            util-linux
+            iproute2
+          ]
+          ++ (lib.optionals encryptionSupport [ openssl ])
+          ++ (lib.optionals ipmiSupport [
+            ipmitool
+            freeipmi
+          ])
+          ++ (lib.optionals ovsSupport [ openvswitch ])
+          ++ (lib.optionals varnishSupport [ varnish ])
+          ++ (lib.optionals postfixSupport [ postfix ])
+          ++ (lib.optionals megaraidSupport [
+            megacli
+            storcli
+          ])
+          ++ (lib.optionals ntpdSupport [ ntp ])
```

```suggestion
          ])
          ++ lib.optional ovsSupport openvswitch
          ++ lib.optional varnishSupport varnish
          ++ lib.optional postfixSupport postfix
          ++ lib.optionals megaraidSupport [
            megacli
            storcli
          ]
          ++ lib.optional ntpdSupport ntp
```
we can minify this a bit